lib/cylc/job_file.py

```diff
@@ -206,9 +206,16 @@ def _write_environment_1(self, handle, job_conf):
             ' '.join(job_conf['namespace_hierarchy']))
         handle.write(
             '\n    export CYLC_TASK_TRY_NUMBER=%s' % job_conf['try_num'])
-        if job_conf['param_var']:
-            for var, val in job_conf['param_var'].items():
-                handle.write('\n    export %s="%s"' % (var, val))
+        # Custom parameter environment variables
+        for var, tmpl in job_conf['param_env_tmpl'].items():
+            try:
+                handle.write('\n    export %s="%s"' % (
+                    var, tmpl % job_conf['param_var']))
+            except (KeyError, TypeError, ValueError):
+                continue
```

Comment:
Is it safe to silently fail?

Reply:
OK. I'll not catch this, and let it fail (which should result in a submit-fail).

(It will annoy users either way - who would like a strict system that is also forgiving!)